Remove serde attribute from chunk field

plustik opened the pull request:
Remove `serde(skip_serializing_if = ...)` attribute from chunk field of `ruma::api::client::r0::message::get_message_events::Response`. This change is necessary because [v1.1 of the Matrix specification](https://spec.matrix.org/v1.1/client-server-api/#get_matrixclientv3roomsroomidmessages) requires the chunk field in the response to **GET /_matrix/client/v3/rooms/{roomId}/messages**. Previously the chunk field was omitted, if it was empty.

Commit:
Remove serde attribute from chunk field

Remove serde(skip_serializing_if = ...) attribute from chunk field of
ruma::api::client::r0::message::get_message_events::Response.
This change is necessary because v1.1 of the Matrix specification
requires the chunk field in the response to GET
/_matrix/client/v3/rooms/{roomId}/messages. Previously the chunk field
was omitted, if it was empty.

jplatte commented:
Thanks for the PR! Two things:

* The start field also got marked as `Required` in v1.1, can you make it non-`Option`al?
* It would be good to update the spec link at the top of the file to point to v1.1 to avoid further confusion.

Commit:
Make start field in get_message_events required

Make the start field of ruma::api::client::r0::message::
get_message_events::Response required by changing its type from
Option<String> to String. This happens in accordance with the updated
specification at
https://spec.matrix.org/v1.1/client-server-api/#get_matrixclientv3roomsroomidmessages.

plustik commented:
> * The start field also got marked as `Required` in v1.1, can you make it non-`Option`al?
> * It would be good to update the spec link at the top of the file to point to v1.1 to avoid further confusion.

I did both now.
Should I also add these changes to CHANGELOG.md? At least the changed type of the start field will propably break the compilation of some projects.

jplatte commented:
I don't think many people read the changelogs and it should be clear from the compiler error what's changed so I think it's not that important.

But also doesn't hurt if you want to add it, just let me know when you consider this ready to merge 🙂

Commit:
Add previous change to CHANGELOG.md

plustik commented:
I've updated the changelog. I think this can be merged now.
